client/src/pages/Chat.tsx

```diff
@@ -132,28 +171,31 @@ export const Chat = () => {
         goBackToChatsPage={goBackToChatsPage}
       />
 
-      <div className="messages">
+      <div className="messages" onScroll={handleScroll}>
         {noMessages ? (
           <span className="no-msg">{companionName ? 'עוד אין הודעות' : 'עוד אין שותף'}</span>
         ) : (
-          messages!
-            .sort((a, b) => {
-              const aDate = new Date(a.date);
-              const bDate = new Date(b.date);
-              return aDate.getTime() - bDate.getTime();
-            })
-            .map((m, index) => (
-              <Message
-                key={index}
-                isSender={m.senderId === user!.uid}
-                content={m.content}
-                messageDate={m.date}
-                messageState={m.status}
-              />
-            ))
-        )}
+            messages!
+              .sort((a, b) => {
+                const aDate = new Date(a.date);
+                const bDate = new Date(b.date);
+                return aDate.getTime() - bDate.getTime();
+              })
+              .map((m, index) => (
+                <Message
+                  key={index}
+                  isSender={m.senderId === user!.uid}
+                  content={m.content}
+                  messageDate={m.date}
+                  messageState={m.status}
+                />
+              ))
+          )}
         <span ref={scrollingRef}></span>
       </div>
+      <div className="has-messages"
+        onClick={scrollDownClick}
+        style={{ display: msgDiff ? "flex" : "none" }}>{msgDiff}</div>
```

Comment:
It is not particularly recommended to insert a style in this way because it is the most rigid. If possible, it is better to avoid and insert style only through the .scss file